.github/workflows/dotnet-ec2-windows-test.yml

```diff
@@ -0,0 +1,334 @@
+              terraform destroy -auto-approve \
+                -var="test_id=${{ env.TESTING_ID }}" 
+          
+              retry_counter=$(($retry_counter+1))
+            else
+              # If deployment succeeded, then exit the loop
+              break
+            fi
+          
+            if [ $retry_counter -eq $max_retry ]; then
+              echo "Max retry reached, failed to deploy terraform and connect to the endpoint. Exiting code"
+              exit 1
+            fi
+          done
+
+      - name: Get the ec2 instance ami id
+        working-directory: terraform/dotnet/ec2/windows
+        run: |
+          echo "EC2_INSTANCE_AMI=$(terraform output ec2_instance_ami)" >> $GITHUB_ENV
+
+      - name: Get SSM outputs
+        working-directory: terraform/dotnet/ec2/windows
+        run: |
+          echo "FRONTEND_SSM_ASSOCIATION_ID=$(terraform output frontend_script_association_id)" >> $GITHUB_ENV
+          echo "REMOTE_SSM_ASSOCIATION__ID=$(terraform output remote_script_association_id)" >> $GITHUB_ENV
+
+      - name: Get the sample app endpoint
+        working-directory: terraform/dotnet/ec2/windows
+        run: |
+          echo "MAIN_SERVICE_ENDPOINT=localhost:8080" >> $GITHUB_ENV
+          echo "REMOTE_SERVICE_IP=$(terraform output sample_app_remote_service_private_ip)" >> $GITHUB_ENV
+          echo "MAIN_SERVICE_INSTANCE_ID=$(terraform output main_service_instance_id)" >> $GITHUB_ENV
+          echo "REMOTE_SERVICE_INSTANCE_ID=$(terraform output remote_service_instance_id)" >> $GITHUB_ENV
+
+      - name: Monitor EC2 instances readiness in Systems Manager
+        run: |
+          max_attempts=60
+          attempt=0
+          interval=10
+
+          check_instance_ready() {
+            local instance_id=$1
+            while [[ $attempt -lt $max_attempts ]]; do
+              state=$(aws ssm describe-instance-information --filters Key=InstanceIds,Values=$instance_id --query "InstanceInformationList[0].PingStatus" --output text)
+              echo "Attempt $((attempt + 1))/$max_attempts: Current SSM state of instance $instance_id: $state"
+
+              if [[ "$state" == "Online" ]]; then
+                echo "Instance $instance_id is 'Online' in Systems Manager and ready."
+                break
+              else
+                echo "Waiting for instance $instance_id to be 'Online' in Systems Manager..."
+                sleep $interval
+                attempt=$((attempt + 1))
+              fi
+            done
+
+            if [[ $attempt -ge $max_attempts ]]; then
+              echo "Timeout reached: Instance $instance_id did not become 'Online' in Systems Manager within expected time."
+              exit 1
+            fi
+          }
+
+          check_instance_ready ${{ env.MAIN_SERVICE_INSTANCE_ID }}
+          check_instance_ready ${{ env.REMOTE_SERVICE_INSTANCE_ID }}
+
+      - name: Repeatedly check SSM Association status until success or max attempts
```

Comment:
This step should be merged with the one above such that the test doesn't spend >10-15 minutes on both of these combined. Spending a potential total worst time of 20 minutes + execution time of the commands is unreasonable, we should fast-fail sooner than that if it's taking that long

Reply:
if the instance is not ready for SSM, SSM association get execute status returns unexpected results: it return success even before execution. 

The current ssm status detection method (next step) should help deal with this problem. It is configured to detect more detailed status, but in order to prevent this weird output from interfering with future engineer debug, the waiting behavior is still retained.

Added max time for this step and the next to kind of solving this concern

Reply:
Discussed offline, deciding to merged steps as suggested.

Reply:
Done